parcel_sourcemap_node/src/lib.rs

```diff
@@ -0,0 +1,581 @@
+        napi_sources_content_array.set_element(
+            source_index as u32,
+            ctx.env.create_string(&source_content[..])?,
+        )?;
+    }
+
+    // Return array
+    return Ok(napi_sources_content_array);
+}
+
+#[js_function]
+fn get_sources_content(ctx: CallContext) -> Result<JsObject> {
+    return _get_sources_content(&ctx);
+}
+
+#[js_function(1)]
+fn get_source_index(ctx: CallContext) -> Result<JsNumber> {
+    let this: JsObject = ctx.this_unchecked();
+    let source_map_instance: &SourceMap = ctx.env.unwrap(&this)?;
+
+    let source = ctx.get::<JsString>(0)?.into_utf8()?;
+    let source_index = source_map_instance.get_source_index(source.as_str()?)?;
+
+    match source_index {
+        Some(i) => {
+            return ctx.env.create_uint32(i);
+        }
+        None => {
+            return ctx.env.create_int32(-1);
+        }
+    }
+}
+
+#[js_function(2)]
+fn set_source_content_by_source(ctx: CallContext) -> Result<JsUndefined> {
+    let this: JsObject = ctx.this_unchecked();
+    let source_map_instance: &mut SourceMap = ctx.env.unwrap(&this)?;
+
+    let source = ctx.get::<JsString>(0)?.into_utf8()?;
+    let source_index: usize = source_map_instance.add_source(source.as_str()?) as usize;
+    let source_content = ctx.get::<JsString>(1)?.into_utf8()?;
+    source_map_instance.set_source_content(source_index, source_content.as_str()?)?;
+
+    return ctx.env.get_undefined();
+}
+
+#[js_function(1)]
+fn get_source_content_by_source(ctx: CallContext) -> Result<JsString> {
+    let this: JsObject = ctx.this_unchecked();
+    let source_map_instance: &mut SourceMap = ctx.env.unwrap(&this)?;
+
+    let source = ctx.get::<JsString>(0)?.into_utf8()?;
+    let source_index = source_map_instance.get_source_index(source.as_str()?)?;
+    match source_index {
+        Some(i) => {
+            let source_content = source_map_instance.get_source_content(i)?;
+            return ctx.env.create_string(source_content);
+        }
+        None => {
+            return ctx.env.create_string("");
+        }
+    }
+}
+
+#[js_function(1)]
+fn add_name(ctx: CallContext) -> Result<JsNumber> {
+    let this: JsObject = ctx.this_unchecked();
+    let source_map_instance: &mut SourceMap = ctx.env.unwrap(&this)?;
+
+    let name = ctx.get::<JsString>(0)?.into_utf8()?;
+    let name_index = source_map_instance.add_name(name.as_str()?);
+    return ctx.env.create_uint32(name_index);
+}
+
+#[js_function(1)]
+fn get_name(ctx: CallContext) -> Result<JsString> {
+    let this: JsObject = ctx.this_unchecked();
+    let source_map_instance: &SourceMap = ctx.env.unwrap(&this)?;
+
+    let name_index = ctx.get::<JsNumber>(0)?.get_uint32()?;
+    match source_map_instance.get_name(name_index) {
+        Ok(name) => {
+            return ctx.env.create_string(name);
+        }
+        Err(_err) => {
+            return ctx.env.create_string("");
```

Comment:
Same with this one. I guess this is ignoring the error atm

Reply:
Same consistency w/ cpp implementation